src/app/ubs-admin/components/ubs-admin-employee/employee-form/employee-form.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { RouterTestingModule } from '@angular/router/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { MatDialogRef, MatDialogModule, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+import { EmployeeFormComponent } from './employee-form.component';
+import { ReactiveFormsModule } from '@angular/forms';
+import { TranslateModule } from '@ngx-translate/core';
+
+describe('EmployeeFormComponent', () => {
+  let component: EmployeeFormComponent;
+  let fixture: ComponentFixture<EmployeeFormComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [EmployeeFormComponent],
+      imports: [MatDialogModule, HttpClientTestingModule, RouterTestingModule, ReactiveFormsModule, TranslateModule.forRoot()],
+      providers: [
+        { provide: MatDialogRef, useValue: {} },
+        { provide: MAT_DIALOG_DATA, useValue: {} }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EmployeeFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('Employee form should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Employee form should be invalid when empty', () => {
+    expect(component.employeeForm.valid).toBeFalsy();
+  });
+
+  it('First name field should be invalid when empty', () => {
+    const firstName = component.employeeForm.controls.firstName;
+    expect(firstName.valid).toBeFalsy();
+  });
+
+  it('First name field should be invalid when user input numbers', () => {
+    const firstName = component.employeeForm.controls.firstName;
+    let errors = {};
+    firstName.setValue('345');
+    errors = firstName.errors || {};
+    expect(Object.keys(errors)).toContain('pattern');
```

Comment:
You shouldn't use that errors variable
If you expect this field to have that error you shouldn't handle the case when it's empty because if its empty it means that your test doesn't work correctly
So please remove that error variable and change the `expect` function call to `expect(firstName.errors.pattern).toBeTruthy();`

Reply:
And the same in some other tests

Reply:
Done